### ISSUE TITLE
Freeing $_ in grep/map causes a double free

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#5038]
Migrated from [rt.perl.org#8527](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=8527) (status was 'resolved')

Searchable as RT8527$

**Comment by p5pRT** [comment]
### From sfink@digital-integrity.com

#### Created by steve@fink.com



perl \-e 'map\{\*\_=5\}1'
Attempt to free unreferenced scalar\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl v5.6.1:

Configured by root at Fri Jan 18 17:19:31 PST 2002.

Summary of my perl5 (revision 5.0 version 6 subversion 1) configuration:
  Platform:
    osname=linux, osvers=2.4.9, archname=i386-linux
    uname='linux foxglove 2.4.9 #4 sat dec 1 17:17:37 pst 2001 i686 unknown '
    config_args='-des -Doptimize=-g -O3 -Dcc=gcc -Dcf_by=Red Hat, Inc. -Dcccdlflags=-fPIC -Dinstallprefix=/usr -Dprefix=/usr -Darchname=i386-linux -Uuselargefiles -Dd_dosuid -Dd_semctl_semun -Di_db -Di_ndbm -Di_gdbm -Di_shadow -Di_syslog -Dman3ext=3pm -Ulocincpth= -Dinc_version_list=5.005 5.005/i386-linux'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=undef usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
  Compiler:
    cc='gcc', ccflags ='-DDEBUGGING -fno-strict-aliasing',
    optimize='-g -O3',
    cppflags='-DDEBUGGING -fno-strict-aliasing'
    ccversion='', gccversion='egcs-2.91.66 19990314/Linux (egcs-1.1.2 release)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=4, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -ldl -lm -lc -lposix -lcrypt -lutil
    perllibs=-lnsl -ldl -lm -lc -lposix -lcrypt -lutil
    libc=/lib/libc-2.1.3.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fPIC', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.6.1:
    /usr/lib/perl5/5.6.1/i386-linux
    /usr/lib/perl5/5.6.1
    /usr/lib/perl5/site_perl/5.6.1/i386-linux
    /usr/lib/perl5/site_perl/5.6.1
    /usr/lib/perl5/site_perl/5.005
    /usr/lib/perl5/site_perl/5.005/i386-linux
    /usr/lib/perl5/site_perl
    .


Environment for perl v5.6.1:
    HOME=/home/sfink
    LANG=en_US
    LANGUAGE (unset)
    LC_ALL=C
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/sfink/down/xref:/bin:/usr/bin:/usr/X11R6/bin:/home/sfink/bin:/usr/local/bin:/usr/local/pilot/bin:/usr/local/samba/bin:/sbin:/usr/sbin:.
    PERL_BADLANG (unset)
    SHELL=/bin/zsh


```
</details>

**Comment by p5pRT** [comment]
### From @schwern

On Wed\, Feb 13\, 2002 at 01&#8203;:25&#8203;:20PM \-0800\, sfink@&#8203;digital\-integrity\.com wrote&#8203;:
> This is a bug report for perl from steve@&#8203;fink\.com\,
> generated with the help of perlbug 1\.33 running under perl v5\.6\.1\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> perl \-e 'map\{\*\_=5\}1'
> Attempt to free unreferenced scalar\.
> 
> \[Please do not change anything below this line\]
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-

$ perl5\.6\.1 \-e 'map\{\*\_=5\}1'
Attempt to free unreferenced scalar\.
$ bleadperl \-e 'map\{\*\_=5\}1'
Segmentation fault

\(gdb\) set args \-e 'map\{\*\_=5\}1'
\(gdb\) run
Starting program&#8203;: /usr/local/bin/bleadperl \-e 'map\{\*\_=5\}1'

Program received signal SIGSEGV\, Segmentation fault\.
0x10093a64 in Perl\_mg\_free \(sv=0x1019e91c\) at mg\.c&#8203;:354
354             if \(vtbl && vtbl\->svt\_free\)
\(gdb\) bt
\#0  0x10093a64 in Perl\_mg\_free \(sv=0x1019e91c\) at mg\.c&#8203;:354
\#1  0x100c3c88 in Perl\_sv\_clear \(sv=0x1019e91c\) at sv\.c&#8203;:4953
\#2  0x100c4464 in Perl\_sv\_free \(sv=0x1019e91c\) at sv\.c&#8203;:5156
\#3  0x100ee1b0 in Perl\_free\_tmps \(\) at scope\.c&#8203;:174
\#4  0x10018964 in perl\_run \(my\_perl=0x1019e008\) at perl\.c&#8203;:1528
\#5  0x10014100 in main \(argc=3\, argv=0x7ffffb44\, env=0x7ffffb54\)
&nbsp;    at perlmain\.c&#8203;:85
\#6  0x0fdd9d30 in \_\_libc\_start\_main \(\) from /lib/libc\.so\.6


\-\- 

Michael G\. Schwern   \<schwern@&#8203;pobox\.com>    http&#8203;://www\.pobox\.com/~schwern/
Perl Quality Assurance        \<perl\-qa@&#8203;perl\.org>           Kwalitee Is Job One
I have this god\-awful need to aquire useless crap\!\!\!

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

sfink@&#8203;digital\-integrity\.com wrote&#8203;:
> 
> This is a bug report for perl from steve@&#8203;fink\.com\,
> generated with the help of perlbug 1\.33 running under perl v5\.6\.1\.
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> perl \-e 'map\{\*\_=5\}1'
> Attempt to free unreferenced scalar\.

Hmm\, doing some random typing on my own perl\, I see this problem occurs
with other things than a bare 5 \(which is interpreted as a symbolic
reference\, IIUC\)&#8203;:
&nbsp;    map \{ \*\_ = \\$5 \} 1;
&nbsp;    map \{ \*\_ = \\$\+ \} 1;
&nbsp;    map \{ \*\_ = \\$\! \} 1;
&nbsp;    map \{ \*\_ = \\$\. \} 1;
&nbsp;    map \{ \*\_ = \\$\# \} 1;
&nbsp;    my $x; map \{ \*\_ = \\$x \} 1;
&nbsp;    map \{ \*\_ = \\$\_ \} 1;
I would even suggest that it occurs with a reference to any scalar\.

How poorly do these others behave on bleadperl?

PS&#8203;: Here's another bug&#8203;:
\[Windows 95\] C&#8203;:\\WINDOWS>perl \-Mstrict=refs \-e"\*foo='bar';print \\$foo\,\\$bar"
SCALAR\(0x17757c8\)SCALAR\(0x17757c8\)

Why doesn't strict refs prevent the assignment of the string to the glob?

\-\- 
From the libwww\-perl changelog\, describing an error fixed in 5\.41&#8203;:
 o  The local/http\.t test actually did try to unlink\("\."\)\.  This was
&nbsp;    very confusing on systems where it succeed\.

**Comment by p5pRT** [comment]
### From @nwc10

On Thu\, Feb 14\, 2002 at 03&#8203;:52&#8203;:59AM \-0500\, Benjamin Goldberg wrote&#8203;:
> sfink@&#8203;digital\-integrity\.com wrote&#8203;:

> > perl \-e 'map\{\*\_=5\}1'
> > Attempt to free unreferenced scalar\.

Also on bleadperl

> Hmm\, doing some random typing on my own perl\, I see this problem occurs
> with other things than a bare 5 \(which is interpreted as a symbolic
> reference\, IIUC\)&#8203;:
>     map \{ \*\_ = \\$5 \} 1;
>     map \{ \*\_ = \\$\+ \} 1;
>     map \{ \*\_ = \\$\! \} 1;
>     map \{ \*\_ = \\$\. \} 1;
>     map \{ \*\_ = \\$\# \} 1;
>     my $x; map \{ \*\_ = \\$x \} 1;
>     map \{ \*\_ = \\$\_ \} 1;
> I would even suggest that it occurs with a reference to any scalar\.
> 
> How poorly do these others behave on bleadperl?

All "Attempt to free unreferenced scalar\." except the last\, which is silent\.
This is on 14766\.

I think you're right about it occurring with a reference to any scalar&#8203;:

nick@&#8203;Bagpuss \[14766\]$ \./perl 
map \{ \*\_ = \\$a \} 1
Attempt to free unreferenced scalar\.

I don't know where to look for a solution to this bug\.

Nicholas Clark
\-\- 
EMCFT http&#8203;://www\.ccl4\.org/~nick/CV\.html

**Comment by p5pRT** [comment]
### From @jhi

On Thu\, Feb 21\, 2002 at 06&#8203;:26&#8203;:12PM \+0000\, Nicholas Clark wrote&#8203;:
> On Thu\, Feb 14\, 2002 at 03&#8203;:52&#8203;:59AM \-0500\, Benjamin Goldberg wrote&#8203;:
> > sfink@&#8203;digital\-integrity\.com wrote&#8203;:
> 
> > > perl \-e 'map\{\*\_=5\}1'
> > > Attempt to free unreferenced scalar\.
> 
> Also on bleadperl

Third Degree shows \(as of 14815\)&#8203;:

scope\.c&#8203;: 789&#8203;: writing invalid heap at byte 0 of 88\-byte block
&nbsp;    Perl\_leave\_scope               libperl\.so\, scope\.c\, line 789
&nbsp;    Perl\_pop\_scope                 libperl\.so\, scope\.c\, line 129
&nbsp;    Perl\_pp\_mapwhile               libperl\.so\, pp\_ctl\.c\, line 849
&nbsp;    Perl\_runops\_debug              libperl\.so\, dump\.c\, line 1392
&nbsp;    S\_run\_body                     libperl\.so\, perl\.c\, line 1601
&nbsp;    perl\_run                       libperl\.so\, perl\.c\, line 1522
&nbsp;    main                           perl\, perlmain\.c\, line 85
&nbsp;    \_\_start                        perl

This block at address 0x140014210 was allocated at&#8203;:
&nbsp;    malloc                         libc\.so
&nbsp;    Perl\_safemalloc                libperl\.so\, util\.c\, line 78
&nbsp;    Perl\_gv\_init                   libperl\.so\, gv\.c\, line 111
&nbsp;    Perl\_gv\_fetchpv                libperl\.so\, gv\.c\, line 746
&nbsp;    S\_init\_main\_stash              libperl\.so\, perl\.c\, line 2656
&nbsp;    S\_parse\_body                   libperl\.so\, perl\.c\, line 1040
&nbsp;    perl\_parse                     libperl\.so\, perl\.c\, line 984
&nbsp;    main                           perl\, perlmain\.c\, line 83
&nbsp;    \_\_start                        perl

This block was freed at&#8203;:
&nbsp;    free                           libc\.so
&nbsp;    Perl\_safefree                  libperl\.so\, util\.c\, line 151
&nbsp;    Perl\_gp\_free                   libperl\.so\, gv\.c\, line 1200
&nbsp;    Perl\_magic\_setglob             libperl\.so\, mg\.c\, line 1560
&nbsp;    Perl\_mg\_set                    libperl\.so\, mg\.c\, line 180
&nbsp;    Perl\_pp\_sassign                libperl\.so\, pp\_hot\.c\, line 115
&nbsp;    Perl\_runops\_debug              libperl\.so\, dump\.c\, line 1392
&nbsp;    S\_run\_body                     libperl\.so\, perl\.c\, line 1601
&nbsp;    perl\_run                       libperl\.so\, perl\.c\, line 1522
&nbsp;    main                           perl\, perlmain\.c\, line 85
&nbsp;    \_\_start                        perl

\-\- 
$jhi\+\+; \# http&#8203;://www\.iki\.fi/jhi/
&nbsp;        \# There is this special biologist word we use for 'stable'\.
&nbsp;        \# It is 'dead'\. \-\- Jack Cohen

**Comment by p5pRT** [comment]
### From @chorny

Same result on 5\.12\.0\.

On Wed Feb 13 05&#8203;:25&#8203;:44 2002\, sfink@&#8203;digital\-integrity\.com wrote&#8203;:

> perl \-e 'map\{\*\_=5\}1'
> Attempt to free unreferenced scalar\.

See discussion at http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=8527

\-\- 
Alexandr Ciornii\, http&#8203;://chorny\.net

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

Same result on 5\.12\.0\.

On Wed Feb 13 05&#8203;:25&#8203;:44 2002\, sfink@&#8203;digital\-integrity\.com wrote&#8203;:

> perl \-e 'map\{\*\_=5\}1'
> Attempt to free unreferenced scalar\.

See discussion at http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=8527

\-\- 
Alexandr Ciornii\, http&#8203;://chorny\.net

**Comment by p5pRT** [comment]
### From @cpansprout

On Wed Feb 13 05&#8203;:25&#8203;:44 2002\, sfink@&#8203;digital\-integrity\.com wrote&#8203;:
> perl \-e 'map\{\*\_=5\}1'
> Attempt to free unreferenced scalar\.

Is this the result of the stack’s not being reference\-counted?

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

On Wed Feb 13 05&#8203;:25&#8203;:44 2002\, sfink@&#8203;digital\-integrity\.com wrote&#8203;:
> perl \-e 'map\{\*\_=5\}1'
> Attempt to free unreferenced scalar\.

Is this the result of the stack’s not being reference\-counted?

**Comment by p5pRT** [comment]
### From @ikegami

On Sun Oct 10 16&#8203;:51&#8203;:32 2010\, sprout wrote&#8203;:
> On Wed Feb 13 05&#8203;:25&#8203;:44 2002\, sfink@&#8203;digital\-integrity\.com wrote&#8203;:
> > perl \-e 'map\{\*\_=5\}1'
> > Attempt to free unreferenced scalar\.
> 
> Is this the result of the stack’s not being reference\-counted?

I think so\. I've added a dependency to the meta ticket just in case\.

**Comment by p5pRT** [comment]
### From @cpansprout

$ perl5\.14\.0 \-e 'grep \{ undef \*\_ \} 1'
Attempt to free unreferenced scalar&#8203;: SV 0x826440\, Perl interpreter&#8203;: 0x800000\.

The same applies to map\, but it’s the same code\.

Can we change the definition of SAVE\_DEFSV?



---
Flags&#8203;:
&nbsp;    category=core
&nbsp;    severity=low
&nbsp;    ack=no


---
Site configuration information for perl 5\.14\.0&#8203;:

Configured by sprout at Wed May 11 13&#8203;:45&#8203;:58 PDT 2011\.

Summary of my perl5 \(revision 5 version 14 subversion 0\) configuration&#8203;:
&nbsp;  Snapshot of&#8203;: eb70bb4a400e88a66c7e10414a2d52b5da4cfd1f
&nbsp;  Platform&#8203;:
&nbsp;    osname=darwin\, osvers=10\.5\.0\, archname=darwin\-thread\-multi\-2level
&nbsp;    uname='darwin pint\.local 10\.5\.0 darwin kernel version 10\.5\.0&#8203;: fri nov 5 23&#8203;:20&#8203;:39 pdt 2010; root&#8203;:xnu\-1504\.9\.17~1release\_i386 i386 '
&nbsp;    config\_args='\-Dusedevel \-de \-Duseithreads \-Doptimize=\-g'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    useithreads=define\, usemultiplicity=define
&nbsp;    useperlio=define\, d\_sfio=undef\, uselargefiles=define\, usesocks=undef
&nbsp;    use64bitint=undef\, use64bitall=undef\, uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'\, ccflags ='\-fno\-common \-DPERL\_DARWIN \-fno\-strict\-aliasing \-pipe \-fstack\-protector \-I/usr/local/include'\,
&nbsp;    optimize='\-g'\,
&nbsp;    cppflags='\-fno\-common \-DPERL\_DARWIN \-fno\-strict\-aliasing \-pipe \-fstack\-protector \-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='4\.2\.1 \(Apple Inc\. build 5664\)'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='env MACOSX\_DEPLOYMENT\_TARGET=10\.3 cc'\, ldflags =' \-fstack\-protector \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /usr/lib
&nbsp;    libs=\-ldbm \-ldl \-lm \-lutil \-lc
&nbsp;    perllibs=\-ldl \-lm \-lutil \-lc
&nbsp;    libc=/usr/lib/libc\.dylib\, so=dylib\, useshrplib=false\, libperl=libperl\.a
&nbsp;    gnulibc\_version=''
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=bundle\, d\_dlsymun=undef\, ccdlflags=' '
&nbsp;    cccdlflags=' '\, lddlflags=' \-bundle \-undefined dynamic\_lookup \-L/usr/local/lib \-fstack\-protector'

Locally applied patches&#8203;:
&nbsp;    RC3



---
@&#8203;INC for perl 5\.14\.0&#8203;:
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.14\.0/darwin\-thread\-multi\-2level
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.14\.0
&nbsp;    /usr/local/lib/perl5/5\.14\.0/darwin\-thread\-multi\-2level
&nbsp;    /usr/local/lib/perl5/5\.14\.0
&nbsp;    /usr/local/lib/perl5/site\_perl
&nbsp;    \.



---
Environment for perl 5\.14\.0&#8203;:
&nbsp;    DYLD\_LIBRARY\_PATH \(unset\)
&nbsp;    HOME=/Users/sprout
&nbsp;    LANG=en\_US\.UTF\-8
&nbsp;    LANGUAGE \(unset\)
&nbsp;    LD\_LIBRARY\_PATH \(unset\)
&nbsp;    LOGDIR \(unset\)
&nbsp;    PATH=/usr/bin&#8203;:/bin&#8203;:/usr/sbin&#8203;:/sbin&#8203;:/usr/local/bin&#8203;:/usr/X11/bin&#8203;:/usr/local/bin
&nbsp;    PERL\_BADLANG \(unset\)
&nbsp;    SHELL=/bin/bash

**Comment by p5pRT** [comment]
### From @cpansprout

Fixed by 55b5114f\.

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'new' to 'resolved'

**Comment by p5pRT** [comment]
### From @cpansprout

On Sun Oct 10 17&#8203;:59&#8203;:54 2010\, ikegami@&#8203;adaelis\.com wrote&#8203;:
> On Sun Oct 10 16&#8203;:51&#8203;:32 2010\, sprout wrote&#8203;:
> > On Wed Feb 13 05&#8203;:25&#8203;:44 2002\, sfink@&#8203;digital\-integrity\.com wrote&#8203;:
> > > perl \-e 'map\{\*\_=5\}1'
> > > Attempt to free unreferenced scalar\.
> > 
> > Is this the result of the stack’s not being reference\-counted?
> 
> I think so\. I've added a dependency to the meta ticket just in case\.

Actually\, it’s not\.  It’s the result of the broken SAVE\_DEFSV
implementation which I fixed in commit 55b5114f\.

It turns out that \#92254 \(which I reported after searching the perl
source for SAVE\_DEFSV\) is a duplicate of this bug\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'

**Comment by p5pRT** [comment]
### From @cpansprout

On Tue Jan 10 12&#8203;:26&#8203;:11 2012\, sprout wrote&#8203;:
> On Sun Oct 10 17&#8203;:59&#8203;:54 2010\, ikegami@&#8203;adaelis\.com wrote&#8203;:
> > On Sun Oct 10 16&#8203;:51&#8203;:32 2010\, sprout wrote&#8203;:
> > > On Wed Feb 13 05&#8203;:25&#8203;:44 2002\, sfink@&#8203;digital\-integrity\.com wrote&#8203;:
> > > > perl \-e 'map\{\*\_=5\}1'
> > > > Attempt to free unreferenced scalar\.
> > > 
> > > Is this the result of the stack’s not being reference\-counted?
> > 
> > I think so\. I've added a dependency to the meta ticket just in case\.
> 
> Actually\, it’s not\.  It’s the result of the broken SAVE\_DEFSV
> implementation which I fixed in commit 55b5114f\.
> 
> It turns out that \#92254 \(which I reported after searching the perl
> source for SAVE\_DEFSV\) is a duplicate of this bug\.
> 

Clarification&#8203;: SAVE\_DEFSV was causing \*\_\{SCALAR\} not to be
reference\-counted\, and operators like undef were not aware of that\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

On Tue Jan 10 12&#8203;:26&#8203;:11 2012\, sprout wrote&#8203;:
> On Sun Oct 10 17&#8203;:59&#8203;:54 2010\, ikegami@&#8203;adaelis\.com wrote&#8203;:
> > On Sun Oct 10 16&#8203;:51&#8203;:32 2010\, sprout wrote&#8203;:
> > > On Wed Feb 13 05&#8203;:25&#8203;:44 2002\, sfink@&#8203;digital\-integrity\.com wrote&#8203;:
> > > > perl \-e 'map\{\*\_=5\}1'
> > > > Attempt to free unreferenced scalar\.
> > > 
> > > Is this the result of the stack’s not being reference\-counted?
> > 
> > I think so\. I've added a dependency to the meta ticket just in case\.
> 
> Actually\, it’s not\.  It’s the result of the broken SAVE\_DEFSV
> implementation which I fixed in commit 55b5114f\.
> 
> It turns out that \#92254 \(which I reported after searching the perl
> source for SAVE\_DEFSV\) is a duplicate of this bug\.
> 

Clarification&#8203;: SAVE\_DEFSV was causing \*\_\{SCALAR\} not to be
reference\-counted\, and operators like undef were not aware of that\.

\-\- 

Father Chrysostomos
